### PR TITLE
Adding `ExpertKnowledge` to `to_dag` in `pgmpy/base/PDAG.py`

### LINKED ISSUE #3487: [ENH] Support `ExpertKnowledge` constraints in `PDAG.to_dag`
**Summary**: `PDAG.to_dag()` currently computes a valid DAG extension of a PDAG but ignores any ExpertKnowledge constraints used during structure learning. As a result, the returned DAG can orient undirected edges in a way that violates required or forbidden edge constraints.

**Proposed Changes**: Extending `PDAG.to_dag()` to accept an optional expert_knowledge parameter.
While selecting removable nodes, ensuring that orienting undirected edges does not:
- violate required_edges_
- violate forbidden_edges_

Since `ExpertKnowledge.fit()` already incorporates search-space and temporal-order constraints into `forbidden_edges_`, no additional handling of `search_space_` is required inside `to_dag()`.

**API Changes**:
```
PDAG.to_dag(expert_knowledge=None)
```
where `expert_knowledge` is an optional fitted `ExpertKnowledge` instance.

**Tests**:

Add a dedicated unit test (`test_expert_knowledge_to_dag`) to verify that `to_dag()` respects:
- required edge constraints
- forbidden edge constraints
- search-space constraints (through `forbidden_edges_` after fitting).

### PR DESCRIPTION
**Pull request by anusa-saha**
**The following checklist is mandatory**

Your PR will be closed if you remove the checklist. Use of LLMs is **strictly forbidden** for any part of this checklist (including for improving language), and will result in a **ban** if we find any use of LLMs.

### Your checklist for this pull request

- [x] Have you followed all the steps from our [Contributing Guide](https://github.com/pgmpy/pgmpy/blob/dev/CONTRIBUTING.md)?
- [x] Does the PR fully address the linked issue and is within its defined scope? If you are still working on the PR, mark it as draft.
- [x] Are all the GitHub Actions checks passing? If not, mark your PR as draft while you fix it.

If you have used AI/LLMs for any assistance, please answer the following questions. Please refer [#2622](https://github.com/pgmpy/pgmpy/pull/2622) for an example of the level of detail we expect:

- [x] Have you reviewed our [AI usage policy](https://github.com/pgmpy/pgmpy/blob/dev/CONTRIBUTING.md#ai-usage-policy)?
- [x] Are you able to fully explain your changes? We expect you to fully understand the algorithm and take full responsibility for any changes in this PR.

- Please list the AI tool(s) used, along with the model and its version used.

[Please replace this with your answer]

- Please describe in **detail** how and what you used AI assistance for? Please outline your whole workflow with the AI tool.

[Please replace this with your answer]

- What steps have you taken to verify that the changes correctly address the issue? What edge cases have you considered? Other than running tests, what else have you verified?

[Please replace this with your answer]

- Have you used AI for generating tests? Can you compress them into a smaller number of tests without losing coverage?

[Please replace this with your answer]

### Issue number(s) that this pull request fixes
- Fixes #3487 

### List of changes to the codebase in this pull request
-

### CODE REVIEW

**Comment by anusa-saha** [comment]
@ankurankan this is exactly what my changes in Adding ExpertKnowledge to to_dag in pgmpy/base/PDAG.py PR (discussed in the meet) does.
Previously, to_dag() followed the standard Dor–Tarsi algorithm. It repeatedly found a removable node  x and once a node was selected, it automatically oriented all of its undirected edges as y(neighbour) -> x. The algorithm only checked whether a node satisfied the Dor-Tarsi conditions (no directed children and the clique condition) and didn't consider any expert knowledge while choosing the orientation.

What I added is a small check before removing the node x. Since removing a node forces all its undirected edges to be oriented as y(neighbor) -> x, I add a  check whether any of those forced orientations would violate the fitted ExpertKnowledge constraints(i.e., required_edges_ and forbidden_edges_). If a forced orientation is the opposite of a required edge or is a forbidden edge, I simply skip that node and continue searching for another valid removable node using continue. This keeps the original Dor-Tarsi algorithm unchanged while ensuring that the returned DAG also respects the expert knowledge whenever possible. Since ExpertKnowledge.fit() already converts temporal-order and search-space constraints into forbidden_edges_, to_dag() only needs to check required_edges_ and forbidden_edges_.

File: pgmpy/base/PDAG.py [hunk 1]
```diff
@@ -254,6 +283,8 @@ def to_dag(self):
         dag.latents = self.latents
 
         pdag = self.copy()
+        required_edges = expert_knowledge.required_edges_ if expert_knowledge is not None else set()
+        forbidden_edges = expert_knowledge.forbidden_edges_ if expert_knowledge is not None else set()
```
Comment (ankurankan) [inline pgmpy/base/PDAG.py]: Let's do the orientation based on the expert knowledge before the actual orientation algorithm is run. Then we can also remove the validation step below as the algorithm doesn't modify any already oriented edges.

**Review by ankurankan — CHANGES_REQUESTED** [review]

**Comment by anusa-saha** [comment]
@ankurankan this is no longer needed as discussed with you in the one to one meet
